Scala 2.10 and 2.11 (new formulas)

qayshp opened the pull request:
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----


My organization (and likely at least a few others) are still using `Scala 2.11` for legacy projects, and noticed when it was removed in [homebrew-versions bb202ed](https://github.com/Homebrew/homebrew-versions/commit/bb202eda53e7a9174caee91ad6c5c7b2d01b61e9). Somewhat surprizingly to me, `scala210` is still in `homebrew-versions`. We have set up our own tap for now, as suggested by @MikeMcQuaid in [this comment](https://github.com/Homebrew/homebrew-versions/commit/ba8b293656ef8fb51eca2d3be022e69ad20277f4#commitcomment-20383905). I'm not sure what analytics are used to determine that a formula is "used by a very small number of people", and if you don't think keeping these formula around is worth it, I would completely understand.

With that said:

This PR adds `Scala 2.11` almost exactly as it was from [`homebrew-versions` `scala211.rb`](https://github.com/Homebrew/homebrew-versions/blob/2bbcc3f2009e55230d41d9c07dd538163350d8fb/scala211.rb).

This PR also updates the `Scala 2.10` formula from [`homebrew-versions` `scala210.rb`](https://github.com/Homebrew/homebrew-versions/blob/838311f4f825af653315c3f8511bbbad1bf6e5c2/scala210.rb) from homebrew-versions from `2.10.5` to `2.10.6` since `2.10` is still in homebrew-versions, and homebrew-versions is now deprecated. This would allow the `scala210` formula to be removed from there.

Both `2.10` and `2.11` formulas are added in this one PR because otherwise `brew audit --strict <formula>` will fail as a result of the `conflicts_with` in each formula pointing at a formula that wouldn't yet exist.

MikeMcQuaid commented:
> I'm not sure what analytics are used to determine that a formula is "used by a very small number of people"

http://docs.brew.sh/Analytics.html. If you or anyone else in your organisation are opting out, I'd advise not doing so.

> This would allow the scala210 formula to be removed from there.

Could you create a homebrew/versions PR to do that, too? That'd be super helpful.

Additionally, add to `tap_migrations.json` in the homebrew/versions PR and the `formula_renames.json` file in this PR.

> if you don't think keeping these formula around is worth it, I would completely understand.

I'm fine with adding them, thanks for the explanation 👍 

Thanks!

MikeMcQuaid commented on Formula/scala@2.10.rb:
```diff
@@ -0,0 +1,68 @@
+  sha256 "54adf583dae6734d66328cafa26d9fa03b8c4cf607e27b9f3915f96e9bcd2d67"
+
+  bottle do
+  end
```
Can remove this empty bottle block.

qayshp commented:
I have done what was suggested above.

- created [PR 1507 on `homebrew/versions`](https://github.com/Homebrew/homebrew-versions/pull/1507) to remove `scala210` and migrate `scala211`.
-- Removes `scala210` formula
-- Marks `scala211` and `scala210` as migrated to `homebrew/core`
- Updated this PR with rebased commits of the `*.rb` files after removing the empty 'bottle' block.
- Updated this PR with `formula_renames.json` for `scala210` and `scala211` migrations.

I tried my best to follow commit message conventions I saw in git blame on the files I was modifying, but please let me know if anything needs to be changed.

Thanks!

qayshp commented:
I hope I addressed everything, but let me know if there are any other changes you would suggest.

MikeMcQuaid commented:
Sorry for the delay, have been away on a work trip, will take a look at this next week.

MikeMcQuaid commented:
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
